spec/message.abnf

```diff
@@ -1,43 +1,46 @@
-message = pattern / complex-message
+message = simple-message / complex-message
 
-complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
+simple-message = [simple-start pattern]
+simple-start = simple-start-char / text-escape / expression
+pattern = *(text-char / text-escape / expression)
 
-declaration = input-declaration / local-declaration
+complex-message = *(declaration [s]) complex-body [s]
```

Comment:
Why do we need `[s]` at the end here?

Reply:
This is the discussion about message trailing whitespace not being invalid, I think?

Reply:
Yeah; this makes is so that `{{Pattern}}\n` is valid. It makes e.g. embedding complex messages in YAML a bit easier, as their trailing newlines don't need to be trimmed.

Note that ` {{Pattern}}` is still invalid, as entering complex mode depends on the first character of the message, and we're not trimming spaces. The difference between the two is that trailing whitespace is harder to see, and doesn't affect the parser.

Reply:
Wait, in `{{Pattern}}\n` the trailing newline is a storage format's concern. Why do you think solving this at the syntax level is appropriate?

Can we discuss this separately from this PR? Why is it in scope?

Reply:
I'd consider this in scope because before this PR this rule was:
```abnf
complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
```
and now it's
```abnf
complex-message = *(declaration [s]) complex-body [s]
```

So removing that terminal `[s]` is a greater change than what's proposed here.

Reply:
> So removing that terminal `[s]` is a greater change than what's proposed here.

I don't think I agree... the current `complex-message = "{{" [s] *(declaration [s]) body [s] "}}"` already does not permit anything after the closing "}}" so e.g. `{{ input {$x} {{foo}} }}` is valid but `{{ input {$x} {{foo}} }}    ` is not. If the scope of this PR is limited to [code-mode-introducer.md](https://github.com/unicode-org/message-format-wg/blob/main/exploration/code-mode-introducer.md), then making `.input {$x} {{foo}}` valid applies but IMO making `.input {$x} {{foo}}    ` valid does not, **especially** since it's making `    .input {$x} {{foo}}    ` (with both leading and trailing whitespace) a valid declaration-free _simple_ message.

I'm not comfortable with the trailing whitespace being invalid, but I'm also not comfortable with whitespace being significant only when it is leading or in an already-established pattern context—and at any rate, changing either more than is necessary to differentiate simple vs. complex messages by how they start would be scope creep.

Reply:
I don't feel like blocking on this will help us, so let's just get rid of the optional trailing whitespace for now and argue about it separately. 

```suggestion
complex-message = *(declaration [s]) complex-body
```